app/scripts/modules/oracle/pipeline/stages/bake/bakeStage.js

```diff
@@ -0,0 +1,207 @@
+    pipelineConfigProvider.registerStage({
+      provides: 'bake',
+      cloudProvider: 'oraclebmcs',
+      label: 'Bake',
+      description: 'Bakes an image in the specified region',
+      templateUrl: require('./bakeStage.html'),
+      executionDetailsUrl: require('./bakeExecutionDetails.html'),
+      executionLabelTemplateUrl: require('core/pipeline/config/stages/bake/BakeExecutionLabel'),
+      defaultTimeoutMs: 60 * 60 * 1000, // 60 minutes
+      validators: [
+        {type: 'requiredField', fieldName: 'baseOs', preventSave: true},
+        {type: 'requiredField', fieldName: 'package', preventSave: true},
+        {type: 'requiredField', fieldName: 'upgrade', preventSave: true},
+        {type: 'requiredField', fieldName: 'region', preventSave: true},
+        {type: 'requiredField', fieldName: 'account', preventSave: true},
+        {type: 'requiredField', fieldName: 'user', preventSave: true},
+        {type: 'requiredField', fieldName: 'cloudProviderType', preventSave: true},
+        {type: 'requiredField', fieldName: 'extended_attributes.subnet_ocid', preventSave: true,
+          message: '<b>Subnet ocid</b> is a required field for Bake stages.'},
+        {type: 'requiredField', fieldName: 'extended_attributes.availability_domain', preventSave: true,
+          message: '<b>Availability Domain</b> is a required field for Bake stages.'}
+      ],
+      restartable: true
+    });
+  })
+  .controller('oraclebmcsBakeStageCtrl', function($scope, bakeryService, accountService, networkReader, subnetReader, $q, authenticationService, settings) {
+
+    const provider = 'oraclebmcs';
+
+    if (!$scope.stage.cloudProvider) {
+      $scope.stage.cloudProvider = provider;
+    }
+
+    if (!$scope.stage) {
+      $scope.stage = {};
+    }
+
+    if (!$scope.stage.extended_attributes) {
+      $scope.stage.extended_attributes = {};
+    }
+
+    if (!$scope.stage.user) {
+      $scope.stage.user = authenticationService.getAuthenticatedUser().name;
+    }
+
+    $scope.account = settings.providers.oraclebmcs.defaults.account;
+
+    function initialize() {
+
+      $scope.viewState.providerSelected = true;
+
+      $q.all({
+        baseOsOptions: bakeryService.getBaseOsOptions(provider),
+        regions: bakeryService.getRegions(provider),
+        availabilityDomains: $scope.getZones(provider),
+        networks: $scope.getNetworks(provider),
+        subnets: $scope.getSubNetworks(provider)
+      }).then(results => {
+
+        if (!$scope.account && $scope.application.defaultCredentials.oraclebmcs) {
+          $scope.account = $scope.application.defaultCredentials.oraclebmcs;
+        }
+        if (results.baseOsOptions.baseImages.length > 0) {
+          $scope.baseOsOptions = results.baseOsOptions;
+        }
+        if (results.regions.length > 0) {
+          $scope.regionOptions = results.regions;
+        }
+        if (results.availabilityDomains && results.availabilityDomains != {}) {
```

Comment:
`{} !== {}` is always true in JS. I would use `Object.keys(results.availabilityDomains).length > 0`.